pkg/v2/tkr/resolver/resolver.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package resolver provides the implementation of the TKR resolver.
+package resolver
+
+import (
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v2/tkr/resolver/data"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v2/tkr/resolver/internal"
+)
+
+// Resolver resolves TKRs and OSImages.
+type Resolver interface {
+	// Resolve returns TKRs and OSImages satisfying query constraints.
+	Resolve(query data.Query) data.Result
+}
+
+// Cache holds TKRs and OSImages to be used by the Resolver.
+type Cache interface {
+	// Add TanzuKubernetesRelease or OSImage to the resolver cache.
+	Add(...interface{})
```

Comment:
A cache in general would also need to provide API for refreshing or invalidating.

Reply:
Coming up with a full fledged cache was not the goal. The goal was to do just enough to provide data to implement the Resolver. 

I admit: a better doc comment definitely wouldn't hurt. Adding a TODO

Reply:
Added `Remove()` method. Also, adding some explanation to the readme.